src/ADNLPProblems/robot_arm_dynamics.jl

```diff
@@ -0,0 +1,87 @@
+using ADNLPModels
+
+function robot_arm_dynamics(;n::Int = 30,L = 4.5, ϵ = 1e-4, type::Val{T} = Val(Float64), kwargs...) where {T}
+N = n
+    n=N+1 # number of timesteps 
+    ϵ = 1e-4
+    # X : vector of variables, of the form : [ρ(t=t1); ρ(t=t2); ... ρ(t=tf), θ(t=t1), ..., then ρ_dot, ..., then ρ_acc, .. ϕ_acc, tf]
+    # There are N+1 values of each 9 variables 
+    # X = [ρ, θ, ϕ, ρ_dot, θ_dot, ϕ_dot, ρ_acc, θ_acc, ϕ_acc, tf]
+
+
+    # objective function 
+    function objective(X)
+        X[end]
+    end
+
+
+    # constraints function 
+    function c(X)
+
+        # position bounds on ρ, θ, ϕ
+        c_ρ = X[1:n]
+        c_θ = X[n+1:2n]
+        c_ϕ = X[2n+1:3n]
+
+
+        # dynamic bounds on ρ_acc, θ_acc, ϕ_acc
+        c_ρ_acc = L*X[6n+1:7n]
+        c_θ_acc = similar(X,n)
+        c_ϕ_acc = similar(X,n)
+        for m=1:n
+            c_θ_acc[m] = X[7n+m] * ((L - X[m])^3 + X[m]^3)/3 * sin(X[2n+m])^2 
+            c_ϕ_acc[m] = X[8n+m] * ((L - X[m])^3 + X[m]^3)/3 
+        end
+
+
+        # tf must be positive
+        c_tf = X[end]
+
+        # Euler's constraints 
+        c_euler = similar(X, 6N)
+        c_euler[1:N] = X[2:n] - X[1:n-1] - X[3n+1:4n-1]*X[end]/n
```

Comment:
Please use dot notation here to avoid allocations: https://docs.julialang.org/en/v1/manual/functions/#man-vectorized